clang/lib/3C/ProgramInfo.cpp

```diff
@@ -402,87 +402,84 @@ void ProgramInfo::exitCompilationUnit() {
   return;
 }
 
-void ProgramInfo::insertIntoExternalFunctionMap(ExternalFunctionMapType &Map,
-                                                const std::string &FuncName,
-                                                FVConstraint *NewC,
-                                                FunctionDecl *FD,
-                                                ASTContext *C) {
-  if (Map.find(FuncName) == Map.end()) {
-    Map[FuncName] = NewC;
-  } else {
-    auto *OldC = Map[FuncName];
-    if (!OldC->hasBody()) {
-      if (NewC->hasBody() ||
-          (OldC->numParams() == 0 && NewC->numParams() != 0)) {
-        NewC->brainTransplant(OldC, *this);
-        Map[FuncName] = NewC;
-      } else {
-        // if the current FV constraint is not a definition?
-        // then merge.
-        std::string ReasonFailed = "";
-        OldC->mergeDeclaration(NewC, *this, ReasonFailed);
-        bool MergingFailed = ReasonFailed != "";
-        if (MergingFailed) {
-          clang::DiagnosticsEngine &DE = C->getDiagnostics();
-          unsigned MergeFailID = DE.getCustomDiagID(
-              DiagnosticsEngine::Fatal, "merging failed for %q0 due to %1");
-          const auto Pointer = reinterpret_cast<intptr_t>(FD);
-          const auto Kind =
-              clang::DiagnosticsEngine::ArgumentKind::ak_nameddecl;
-          auto DiagBuilder = DE.Report(FD->getLocation(), MergeFailID);
-          DiagBuilder.AddTaggedVal(Pointer, Kind);
-          DiagBuilder.AddString(ReasonFailed);
-        }
-        if (MergingFailed) {
-          // Kill the process and stop conversion
-          // Without this code here, 3C simply ignores this pair of functions
-          // and converts the rest of the files as it will (in semi-compliance
-          // with Mike's (2) listed on the original issue (#283)
-          exit(1);
-        }
-      }
-    } else if (NewC->hasBody()) {
-      clang::DiagnosticsEngine &DE = C->getDiagnostics();
-      unsigned DuplicateDefinitionsID = DE.getCustomDiagID(
-          DiagnosticsEngine::Fatal, "duplicate definition for function %0");
-      DE.Report(FD->getLocation(), DuplicateDefinitionsID).AddString(FuncName);
-      exit(1);
-    } else {
-      // The old constraint has a body, but we've encountered another prototype
-      // for the function.
-      assert(OldC->hasBody() && !NewC->hasBody());
-      // By transplanting the atoms of OldC into NewC, we ensure that any
-      // constraints applied to NewC later on constrain the atoms of OldC.
-      NewC->brainTransplant(OldC, *this);
+void ProgramInfo::insertNewFVConstraint(FunctionDecl *FD, FVConstraint *NewC,
```

Comment:
The old functions were removed from `ProgramInfo.h`; why was this new function not added there?

Reply:
It always existed as the entry point and branched into the two functions now removed. (But since 2/3 of then were minor code then branch, I combined them all)